### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by mbullington**
- Upgrade to Expo 39, remove `android/` and `ios/` folders since Expo makes them more or less useless.
- Stop using `AsyncData` and `mobx`. We originally did this for frontend parity, but we can simplify a lot and the frontend can eventually follow. Now we use `ChangeNotifier` (I wrote my own, but same API as Flutter) and `swr`, which is awesome and so simple. We get a ton of stuff out of the box.
- Add a linting configuration so our code looks the same. I just reused our open-source one from work (DJI). If you install the ESLint plugin in whatever editor, it should work great. Future ticket can be adding a GitHub Action to make compliance with ESLint mandatory.
- Small things here and there.

### CODE REVIEW

**Review by rsquared226 — CHANGES_REQUESTED** [review]
Update is super great! Simplifies a lot of stuff and it seems to be working great for the most part but I'm getting some warnings (on Android). I put the bolded part of the warning message below, but I can give more details if you need it.

First one happens right when the app is done building and just launched, it's about there not being a native splash screen:
`[Unhandled promise rejection: Error: No native splash screen registered for provided activity. Please configure your application's main Activity to call 'SplashScreen.show' (https://github.com/expo/expo/tree/master/packages/expo-splash-screen#-configure-android).]`

Second one happens after being on the account page for a little, but it might be unrelated to that and could've just popped up then out of coincidence. It's about a timer being set for a long time:
`Setting a timer for a long period of time, i.e. multiple minutes, is a performance and correctness issue on Android as it keeps the timer module awake, and timers can only be called when the app is in the foreground. See https://github.com/facebook/react-native/issues/12981 for more info.
(Saw setTimeout with duration 90000ms)`

I'm also getting this from the console, I'm not sure which console log it's from lol (with whitespace, not sure why it's being removed):
`Object {
  "data": null,
  "error": undefined,
  "isValidating": false,
  "mutate": [Function anonymous],
  "revalidate": [Function anonymous],
}`

Also, is it just me or is the space between the logo and title bigger?
image

File: package.json [hunk 1]
```diff
@@ -1,10 +1,15 @@
 {
+  "name": "hackpsu_app",
+  "private": true,
+  "contributors": [
+    "Rahul Ramkumar",
+    "Michael Bullington",
+    "Ali Malik",
+    "Jonathan Tran",
+    "Tommy Zhao"
+  ],
```
[the remark this answers is not in the document]
Comment (rsquared226) [reply package.json:10]: Wait put your name at the top lol, you're carrying

[the remark this answers is not in the document]
Comment (mbullington) [reply package.json]: 💯

**Comment by mbullington** [comment]
> Update is super great! Simplifies a lot of stuff and it seems to be working great for the most part but I'm getting some warnings (on Android). I put the bolded part of the warning message below, but I can give more details if you need it.
>
> First one happens right when the app is done building and just launched, it's about there not being a native splash screen:
> `[Unhandled promise rejection: Error: No native splash screen registered for provided activity. Please configure your application's main Activity to call 'SplashScreen.show' (https://github.com/expo/expo/tree/master/packages/expo-splash-screen#-configure-android).]`

So apparently this is an Expo issue. Would you be okay just letting it go until a fix? https://github.com/expo/expo/issues/10263                            
 
> Second one happens after being on the account page for a little, but it might be unrelated to that and could've just popped up then out of coincidence. It's about a timer being set for a long time. It's popped up about 3 times when I just left my phone on:
> `Setting a timer for a long period of time, i.e. multiple minutes, is a performance and correctness issue on Android as it keeps the timer module awake, and timers can only be called when the app is in the foreground. See https://github.com/facebook/react-native/issues/12981 for more info. (Saw setTimeout with duration 90000ms)`

Not sure about this one, we don't use `setTimeout` anywhere.

> I'm also getting this from the console, I'm not sure which console log it's from lol (with whitespace, not sure why it's being removed):
> `Object { "data": null, "error": undefined, "isValidating": false, "mutate": [Function anonymous], "revalidate": [Function anonymous], }`

Fixed.

> Also, is it just me or is the space between the logo and title bigger?
> image

Don't think so but shrunk anyway.

**Review by rsquared226 — CHANGES_REQUESTED** [review]
> So apparently this is an Expo issue. Would you be okay just letting it go until a fix? expo/expo#10263

Yeah that works

> Not sure about this one, we don't use setTimeout anywhere.

I looked into it and it turns out to be an issue with React Native and Firebase. We could either ignore the warning or use a solution like this: https://github.com/firebase/firebase-js-sdk/issues/97#issuecomment-427512040
If we go with that file, where would we keep it and call it?

Also, I'm getting a lot of errors from eslint because of line endings. I found the solution though: https://stackoverflow.com/q/53516594
Would we need to add that to your dji eslint library?

**Comment by mbullington** [comment]
should be good now, can you try? thanks!!

**Review by rsquared226 — APPROVED** [review]
It works! The splashscreen warning got fixed just this morning, and the setting a timer fix helped prevent the bug from showing up a lot but it still happens after sometime, I just ignored it for now.
You popped off

**Review by jaytran505 — APPROVED** [review]

**Pull request merged**